scripts/deploy.sh

```diff
@@ -22,10 +22,12 @@ export REGISTRY=quay.io/kubernetes-service-catalog/
 
 docker login -e="${QUAY_EMAIL}" -u "${QUAY_USERNAME}" -p "${QUAY_PASSWORD}" quay.io
 
-if [[ -n "${TRAVIS_TAG}" ]]; then
+if [[ "${TRAVIS_TAG}" =~ ^v[0-9]\.[0-9]\.[0-9][a-z]*$ ]]; then
     echo "Pushing images with tag ${TRAVIS_TAG}."
     VERSION="${TRAVIS_TAG}" make push
-elif [[ "${TRAVIS_PULL_REQUEST}" == "false" && "${TRAVIS_BRANCH}" == "master" ]]; then
+elif [[ "${TRAVIS_BRANCH}" == "master" ]]; then
```

Comment:
@krancour iirc on a PR, `TRAVIS_BRANCH` indicates the branch that the PR will merge into.

According to https://docs.travis-ci.com/user/environment-variables/#Default-Environment-Variables:

```console
TRAVIS_BRANCH:
for push builds, or builds not triggered by a pull request, this is the name of the branch.
for builds triggered by a pull request this is the name of the branch targeted by the pull request.
Note that for tags, git does not store the branch from which a commit was tagged.
```